spec/system/request_tracker_spec.rb

```diff
@@ -11,12 +11,15 @@
     CachedCounting.reset
     ApplicationRequest.disable
     CachedCounting.disable
+    DiscourseEvent.all_off(:page_visited)
   end
 
   describe "pageviews" do
     it "tracks an anonymous visit correctly" do
-      visit "/"
+      events = []
+      DiscourseEvent.on(:page_visited) { |payload| events << payload }
```

Comment:
I think we should be able to use [DiscourseEvent.track_events](https://github.com/discourse/discourse/blob/58a8668d047ba3e93e709570b6163a2a9d7da495/spec/support/discourse_event_helper.rb#L13) here instead of having to register a custom event.